### PR TITLE
Add dockerstats receiver to OpenTelemetry collector

### PR DESCRIPTION
**Pull request by Sanrag**
This receiver will periodically query the docker API to fetch container stats and export them as metrics. This change only adds container mem usage/limit metrics, others to follow.

Requires docker unix socket (generally `/var/run/docker.sock`) to be accessible to the collector.

### CODE REVIEW

File: opentelemetry_collector/receiver/dockerstats/scraper.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+			}
+		}
+	}()
+}
+
+func (s *scraper) stop() {
+	s.done <- true
+}
+
+func (s *scraper) export() {
```
Comment (jiggoha) [inline opentelemetry_collector/receiver/dockerstats/scraper.go:89]: Should we include a log line saying we're starting an export here? I'm thinking about debugging the case where none or not all of the containers are showing up.

File: opentelemetry_collector/receiver/dockerstats/scraper.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	if err != nil {
+		glog.Warningf("Failed to get docker container list: %v", err)
+		return
+	}
+
+	var metrics []*mpb.Metric
+	for _, container := range containers {
+		var name string
+		if len(container.Names) > 0 {
+			name = strings.TrimPrefix(container.Names[0], "/")
```
Comment (jiggoha) [inline opentelemetry_collector/receiver/dockerstats/scraper.go:103]: Can you add a comment explaining the trimming?

File: opentelemetry_collector/receiver/dockerstats/scraper.go (hunk 1 above)
Comment (Sanrag) [reply opentelemetry_collector/receiver/dockerstats/scraper.go:89]: I've a minor concern with log spam -- any log line here would be printed every 30s. Can you explain more what particular scenario would you want to be able to debug? Maybe there's a better place for such a line.

Comment (jiggoha) [reply opentelemetry_collector/receiver/dockerstats/scraper.go:89]: If we don't see any metrics showing up, then something with docker or OpenTelemetry is up. How can we tell if OpenTelemetry is the problem?

File: opentelemetry_collector/receiver/dockerstats/scraper.go (hunk 2 above)
Comment (Sanrag) [reply opentelemetry_collector/receiver/dockerstats/scraper.go:106]: Done

File: opentelemetry_collector/receiver/dockerstats/scraper.go (hunk 1 above)
Comment (Sanrag) [reply opentelemetry_collector/receiver/dockerstats/scraper.go:89]: I see what you're saying. Added an info log line.

**Pull request merged**